### DISCUSSION TITLE
Is there a way to filter out files in a sub-folder?

### DISCUSSION THREAD

**Discussion by kquinsland** [RafaelGB/obsidian-db-folder#155] [Q&A]
Thank you for writing this plugin. I have been waiting for this type of function to make it to Obsidian and I am _very_ grateful that it's now here!.

---

Let's say that I have a directory structure that looks like this:

```
.
└── vaultRoot/
    ├── projects/
    │   ├── someProjectNamehere/
    │   │   ├── _files/
    │   │   │   ├── someDocument.pdf
    │   │   │   └── someDIagram.png
    │   │   ├── index.md
    │   │   ├── someOtherNoteRelatedToTheProject.md
    │   │   ├── randomThoughts.md
    │   │   ├── webClipping.md
    │   │   └── otherExampleMiscNoteHere.md
    │   └── someOtherProjectNamehere/
    │       ├── _files
    │       └── index.md
    └── archive/
        └── 2019/
            └── someOldProjectNameHere/
                ├── _files
                └── index.md
```

How can I make sure that **only** `$projectNameHere/index.md` files show up in the view?  I don't need files like ` webClipping.md` to show up and a really want everything below `archive/` or any `_files/` dir to not show up.

I have tried playing with the "add new filter" in the settings for the db folder (`file.path contains archive`) but there is no way to do a `NOT` operation there (as far as I can tell. Is there a way to view the queries that this plugin feeds into the data view engine?)

**Comment by RafaelGB** [comment]
Hi @kquinsland , Recently Dataview was updated to support queries with its API as users wrote them. I'm working to support it as a new source option.
Hope that will cover that kind of filtering. Currently the no-code filters are directed to fields but not subfields yet...

The idea of your filter will be `file.name="index"` but just file field is available with my plugin. Let me work on it =)

**Reply by kquinsland** [reply]
> I'm working to support it as a new source option.

Can you explain a bit more about this? Does this mean that I can specify my own query for a column?

>  just file field is available with my plugin.

Gotcha. That would explain why the query does not work :P


Are you planning on adding any logging or similar that would allow me to examine _exactly_ what gets fed into the dataview API? This would be very helpful for debugging things in general.

**Reply by RafaelGB** [reply]
There is a developer section when you can enable logging by levels.

Yes, you will can to use your own queries. Currently I am using the dataviewjs pages and the filters is a feature of the plugin, not of dataview

```ts
/**
 * With the use of Dataview and the folder path, we can obtain an array of rows
 * @param folderPath 
 * @returns 
 */
export async function adapterTFilesToRows(folderPath: string, columns: TableColumn[], dbYaml: DatabaseYaml): Promise<Array<RowDataType>> {
  LOGGER.debug(`=> adapterTFilesToRows.  folderPath:${folderPath}`);
  const rows: Array<RowDataType> = [];
  let id = 0;

  let folderFiles = await sourceDataviewPages(folderPath, dbYaml);
  folderFiles.where(p => !p[DatabaseCore.FRONTMATTER_KEY]);
  // Config filters asociated with the database
  if (dbYaml.filters) {
    folderFiles = folderFiles.where(p => DataviewService.filter(dbYaml.filters, p));
  }
  folderFiles.map((page) => {
    const noteInfo = new NoteInfo(page, ++id);
    rows.push(noteInfo.getRowDataType(columns));
  });

  LOGGER.debug(`<= adapterTFilesToRows.  number of rows:${rows.length}`);
  return rows;
}

export async function sourceDataviewPages(folderPath: string, dbYaml: DatabaseYaml): Promise<DataArray<Record<string, Literal>>> {
  switch (dbYaml.config.source_data) {
    case SourceDataTypes.TAG:
      return DataviewService.getDataviewAPI().pages(`#${dbYaml.config.source_form_result}`);
    case SourceDataTypes.INCOMING_LINK:
      return DataviewService.getDataviewAPI().pages(`[[${dbYaml.config.source_form_result}]]`);
    case SourceDataTypes.OUTGOING_LINK:
      return DataviewService.getDataviewAPI().pages(`outgoing([[${dbYaml.config.source_form_result}]])`);
    case SourceDataTypes.QUERY:
      // *************Working on THIS***************
      return obtainQueryResult(dbYaml.config.source_form_result);
    default:
      return DataviewService.getDataviewAPI().pages(`"${folderPath}"`);
  }
}
```

**Comment by kquinsland** [comment]
Just wanted to provide an update!

The recent 1.8.0 release which supports setting type `query` worked. I added `FROM "Projects" WHERE file.name = "index"` and now the rows in the table are as expected. 


However, _now_ I have a table that looks like this:


| File | Project | 
| -- | -- |
| "index" | someProjectNamehere |
| "index" | someOtherProjectNamehere |

... ETC.

Is there a way to Hide the `FIle` column and make the `Project` column the link to file? AS I understand it, the new `query` type only lets me supply arbitrary text to the DV query but only  at the "end" after the columns have been setup.

**Reply by RafaelGB** [reply]
You can add a link as a value in your Project field (text type column)

About hide the file it's already an issue into the roadmap with that purpose but its not ready yet

https://github.com/RafaelGB/obsidian-db-folder/issues/174

**Reply by kquinsland** [reply]
> You can add a link as a value in your Project field (text type column)

Can you clarify what you mean by this?  When I click on the `Settings` for the column, I get a toggle for inline/front matter and media support.

I created a new `selfLink: [[projects/someProjectNamehere/index|someProjectNamehere]]` in front matter but the new `selfLink` column just renders the _literal_ text ` [[projects/someProjectNamehere/index|someProjectNamehere]]` and not as [someProjectNamehere](projects/someProjectNamehere/index.md)

**Reply by RafaelGB** [reply]
hmm with the column as inline (a * should appear to the right of the label) the link should works. Modify something inside the cell to apply the change of type
